src/lib/components/custom/code.svelte

```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+	import { Copy, Checkmark } from 'carbon-icons-svelte'
+	import Button from '$lib/components/ui/button.svelte'
+
+	import { Highlight, HighlightSvelte } from 'svelte-highlight'
+	import { css } from 'svelte-highlight/languages'
+	import type { HTMLAttributes } from 'svelte/elements'
+	import '$lib/components/custom/highlight-diete.css'
+
+	type Dimension = 'default' | 'large' | 'compact' | 'small'
+	interface Props extends HTMLAttributes<HTMLElement> {
+		dimension?: Dimension
+		class?: string
+		code: string
+		language: 'css' | 'svelte'
+	}
+	let {
+		dimension = 'default',
+		class: classProp = '',
+		code,
+		language,
+		...restProps
+	}: Props = $props()
+
+	let buttonIcon = $state('copy')
+
+	function onclick() {
+		navigator.clipboard.writeText(code)
+		buttonIcon = 'checkmark'
+		setTimeout(() => (buttonIcon = 'copy'), 2000)
```

Comment:
Not a big issue, but I'd suggest to clear the timeout on each click. This will make it much nicer when I click several times with some delay.

Reply:
Done.